backend/main/shared/snapshot-flow.go

```diff
@@ -207,6 +214,32 @@ func (c *SnapshotClient) GetLatestSnapshot(contract Contract) (*Snapshot, error)
 	return &snapshot, nil
 }
 
+func (c *SnapshotClient) AddFungibleToken(addr, name, path string) (error) {
+	if c.bypass() {
+		return nil
+	}
+
+	url := fmt.Sprintf(`%s/add-fungible-token`, c.BaseURL)
+
+	payload := FungibleTokenContract{
+		ContractAddress: addr,
+		ContractName: name,
+		PublicCapabilityPath: path,
+	}
+	req, err := c.setPostRequest(url, payload)
```

Comment:
inside `setPostRequest` you can call `setRequestMethod` and pass in `"POST"` as the method arg, this will remove a bit of repeating logic.

Reply:
The reason that I have done it this way is that I need to add a body to the post request, and the setRequestMethod doesn't take a body as a parameter. Since optional arguments is a sort of go-anti pattern, I created a function that accepts a struct and sets the body. Is there a way that we can make this more generic?